nibabel/tests/test_volumeutils.py

```diff
@@ -692,9 +692,15 @@ def test_a2f_nan2zero_range():
             write_return(arr_no_nan, fobj, np.int8, intercept=129)
         # OK with nan2zero false, but we get whatever nan casts to
         with pytest.warns(warn_type) if warn_type else error_warnings():
-            nan_cast = np.array(np.nan, dtype=dt).astype(np.int8)
+            # XXX NP1.24
+            # Casting nan to int will produce a RuntimeWarning in numpy 1.24
+            # Change to expecting this warning when this becomes our minimum
+            with np.errstate(invalid='ignore'):
+                nan_cast = np.array(np.nan, dtype=dt).astype(np.int8)
         with pytest.warns(warn_type) if warn_type else error_warnings():
-            back_arr = write_return(arr, fobj, np.int8, intercept=129, nan2zero=False)
+            # XXX NP1.24 - expect RuntimeWarning
```

Comment:
What do you think about a Numpy version check here and checking for RunTime error, so we don't forget?

Reply:
Was trying to avoid it at 3am, but I guess it does make sense to go ahead and put the anticipated final test in now. Using Version('1.24.dev0+239') to match https://github.com/numpy/numpy/commit/abc26a351e14c890bdd67785ed4f3c60c637b0fa.

Reply:
Sorry to make you do that!